.github/workflows/PR-checks.yml

```
@@ -0,0 +1,90 @@
+name: PR checks
+
+on:
+  pull_request:
+    branches: [ "main", "development" ]
+
+jobs:
+  lint:
+    permissions:
+      checks: write
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [ 18.x ]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - run: npm ci
+      - run: npm run lint -- --output-file eslint_report.json --format json
+
+      - name: Annotate Code Linting Results
+        uses: ataylorme/eslint-annotate-action@1.2.0
+        if: always()
+        with:
+          repo-token: '${{ secrets.GITHUB_TOKEN }}'
+          report-json: 'eslint_report.json'
+          fail-on-warning: 'true'
+
+  coverage:
+    permissions:
+      pull-requests: write
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [ 18.x ]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+
+      - run: npm ci
+      - name: Run tests
+        run: 'npm run test -- --coverage --coverageReporters json-summary | tee ./coverage.txt'
```

Since we are now using jest as a standalone package, this option is now needed.
```suggestion
        run: 'npm run test -- --coverage --coverageReporters json-summary  --coverageReporters text | tee ./coverage.txt'
```